Add enrollment hints

geonnave opened the pull request:
Addresses #7. Work in progress.

chrysn reviewed (commented):
The voucher details are something I care about, and that text looks good to me.

gselander commented:
I read through the changes, some minor comments: 

* If could be good to clarify that “OPAQUE_INFO” is opaque when viewed as information element in EDHOC. It is opaque to V but the application in U and W can read the contents.
* “The hint is an optional field in EAD_1, and it must be encrypted.”
Clarify that u_hint is included in OPAQUE_INFO.
* Similarly, clarify that v_hint in included in REJECT_INFO
* Nit: All words in upper case in titles: "Enrollment hints"->"Enrollment Hints", etc.

gselander commented:
One more comment: 

"W generates error_content and transfers it to V via the secure connection."

We should explain how this message is part of the VREQ/VRES protocol. This is not necessary an error from the point of view of that protocol. If we don't have a quick fix, make a separate issue.

geonnave commented:
Thank you @chrysn and @gselander for your comments.

Note that the reviewed text included more commits than just the hints. I just rebased it from master to include just the hints' update.

Finally, @gselander I just addressed the comments that applied to the hints section. Will address the remaining ones in a separate issue.
